pkg/types/baremetal/validation/platform.go

```diff
@@ -393,6 +393,21 @@ func ValidateProvisioning(p *baremetal.Platform, n *types.Networking, fldPath *f
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("libvirtURI"), p.LibvirtURI, err.Error()))
 	}
 
+	if p.BootstrapOSImage != "" {
+		if err := validateOSImageURI(p.BootstrapOSImage); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("bootstrapOSImage"), p.BootstrapOSImage, err.Error()))
+		}
+	}
+	if p.ClusterOSImage != "" {
+		if err := validateOSImageURI(p.ClusterOSImage); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("clusterOSImage"), p.ClusterOSImage, err.Error()))
+		}
+		// The URL will contain a checksum after the '?' so we can just check for the suffix followed by a '?'.
+		if !strings.Contains(p.ClusterOSImage, "qcow2.gz?") && !strings.Contains(p.ClusterOSImage, "qcow2.xz?") {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("clusterOSImage"), p.ClusterOSImage, "cluster baremetal image must end in .qcow2.gz or .qcow2.xz"))
+		}
```

Comment:
Why do we only need to do this validation for the clusterOSImage, vs for both e.g in validateOSImageURI?

Reply:
Hmm, good point.  My intention was to catch this error before it is checked in the same why by the downloader container and it fails there.  I haven't traced BootstrapOSImage through to know it needs this validation.  I'll try applying it to both and see that it works.

Reply:
I think we need it for both - the same downloader container is used for the bootstrap image:

https://github.com/openshift/installer/blob/master/data/data/bootstrap/baremetal/files/usr/local/bin/startironic.sh.template#L92